File: 3_Descriptive_Statistics.R
# Goodreads_Books 
# Descriptive Statistics
# Haifaa Alzahrani
# 21/10/2020

######################################################################################################################

# Theoretical Probability Distribution ----
# Empirical frequency distribution (on the sample = dataset)
# It is distributions of a variable that we observe in a given sample.

# Overview 
summary(dataset)

# Average rating distribution
ggplot(dataset,aes(x=average_rating)) +      
  geom_density(alpha= 0.5,fill='#8dd3c7')+ 
  geom_vline(aes(xintercept=mean(average_rating)), linetype='dashed')+
  geom_text(aes(x=4.1, y=.8, label='mean'), size=3)+
  ggtitle('Average Rating Destribution')

# Ratings count distribution
dataset %>% 
  ggplot(aes(x=ratings_count)) + 
  geom_histogram(bins = 30, fill = "#bebada")+ 
  ggtitle('Ratings Count Destribution')

# Review count distribution
dataset %>% 
  ggplot(aes(x= text_reviews_count)) + 
  geom_histogram(bins=30, fill = "#fdb462")+ 
  ggtitle('Review Count Destribution')

# Page Number distribution
dataset %>% 
  ggplot(aes(x=num_pages)) + 
  geom_histogram(bins = 30, fill = "#bebada")+ 
  ggtitle('Page Number Destribution')

######################################################################################################################

# Understanding the relationships among the variables ----

# Which factor is most highly-correlated with the average rating of the book?

# Do the awards correlate with the average rating of the book?

# Is the number of pages and rating counts correlated?

File: Python/3_Descriptive_Statistics.py
# Descriptive Statistics
# date: 21/10/2020
# Haifaa Alzahrani

# %%
# Theoretical Probability Distribution
# It is empirical frequency distribution (on the sample = dataset),
# which reflects the distributions of a variable that we observe in a given sample.

# %%
# Import packages
import math
import pandas as pd # For DataFrame 
import numpy as np # Array and numerical processing
import matplotlib.pyplot as plt # Low level plotting
import seaborn as sns # High level Plotting

# %%
# Read the dataset
books = pd.read_csv("~/MISK_DSI/R/DS_Research_GoodreadsBooks/Data/full_books_Python.csv")

# Overview on descriptive statistics
books.describe()

# %%
# Average rating distribution.
sns.distplot(books['average_rating'])

# %% 
# Ratings count distribution.
sns.distplot(books['ratings_count'])


# %% 
# Review count distribution
sns.distplot(books['text_reviews_count'])

# %% 
# Page Number distribution
sns.distplot(books['num_pages'])

# %%
books.head()
